web/modules/custom/dpl_patron_page/src/Plugin/Block/PatronPageBlock.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Drupal\dpl_patron_page\Plugin\Block;
+
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\dpl_react_apps\Controller\DplReactAppsController;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\dpl_library_agency\Branch\BranchRepositoryInterface;
+use Drupal\dpl_library_agency\BranchSettings;
+
+/**
+ * Provides patron page.
+ *
+ * @Block(
+ *   id = "dpl_patron_page_block",
+ *   admin_label = "Patron page"
+ * )
+ */
+class PatronPageBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Drupal config factory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  private ConfigFactoryInterface $configFactory;
+
+  /**
+   * PatronPageBlock constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   Drupal config factory to get FBS and Publizon settings.
+   * @param \Drupal\dpl_library_agency\BranchSettings $branchSettings
+   *   Branch settings.
+   * @param \Drupal\dpl_library_agency\Branch\BranchRepositoryInterface $branchRepository
+   *   Branch repository.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $configFactory, private BranchSettings $branchSettings, private BranchRepositoryInterface $branchRepository) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configuration = $configuration;
+    $this->configFactory = $configFactory;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('dpl_library_agency.branch_settings'),
+      $container->get('dpl_library_agency.branch.repository'),
+    );
+  }
+
+  /**
+   * Checks whether the library has enabled text messages.
+   *
+   * @return string
+   *   True if enabled, false if disabled.
+   */
+  private function textNotificationsEnabled(): string {
+    $patron_page_settings = $this->configFactory->get('patron_page.settings');
+    return empty($patron_page_settings->get('text_notifications_enabled')) ? 'false' : 'true';
+  }
```

Comment:
We have been using "1"/"0" inother cases. I think we should stick to one way of repesenting boolean in config

Reply:
```suggestion
    return empty($patron_page_settings->get('text_notifications_enabled')) ? '0' : '1';
```

wie so?

Reply:
Si